src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerParameterSymbol.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+#nullable enable
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed class FunctionPointerParameterSymbol : ParameterSymbol
+    {
+        private readonly FunctionPointerMethodSymbol _containingSymbol;
+
+        public FunctionPointerParameterSymbol(TypeWithAnnotations typeWithAnnotations, RefKind refKind, int ordinal, FunctionPointerMethodSymbol containingSymbol, ImmutableArray<CustomModifier> refCustomModifiers)
+        {
+            TypeWithAnnotations = typeWithAnnotations;
+            RefKind = refKind;
+            Ordinal = ordinal;
+            _containingSymbol = containingSymbol;
+            RefCustomModifiers = refCustomModifiers;
+        }
+
+        public override TypeWithAnnotations TypeWithAnnotations { get; }
+        public override RefKind RefKind { get; }
+        public override int Ordinal { get; }
+        public override Symbol ContainingSymbol => _containingSymbol;
+        public override ImmutableArray<CustomModifier> RefCustomModifiers { get; }
+
+        public override bool Equals(Symbol other, TypeCompareKind compareKind)
+        {
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            if (!(other is FunctionPointerParameterSymbol param))
+            {
+                return false;
+            }
+
+            return Equals(param, compareKind, isValueTypeOverride: null);
+        }
+
+        internal bool Equals(FunctionPointerParameterSymbol other, TypeCompareKind compareKind, IReadOnlyDictionary<TypeParameterSymbol, bool>? isValueTypeOverride)
+            => other.Ordinal == Ordinal
+               && _containingSymbol.Equals(other._containingSymbol, compareKind, isValueTypeOverride);
+
+        internal bool MethodEqualityChecks(FunctionPointerParameterSymbol other, TypeCompareKind compareKind, IReadOnlyDictionary<TypeParameterSymbol, bool>? isValueTypeOverride)
+            => RefKind == other.RefKind
+               && ((compareKind & TypeCompareKind.IgnoreCustomModifiersAndArraySizesAndLowerBounds) == 0
+                    ? RefCustomModifiers.Equals(other.RefCustomModifiers)
```

Comment:
>Equals [](start = 41, length = 6)

Is this a value or a reference comparison?

Reply:
Do we have a test that relies on value equality of modifiers?

---
In reply to: [363539956](https://github.com/dotnet/roslyn/pull/40431#discussion_r363539956) [](ancestors = 363539956)